File: MyFirstAPP/大众点评/发现/DZDiscoverController.m
//
//  DZDiscoverController.m
//  Shop
//
//  Created by dengwei on 15/12/1.
//  Copyright (c) 2015年 dengw. All rights reserved.
//

#import "DZDiscoverController.h"

@interface DZDiscoverController ()

@end

@implementation DZDiscoverController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self buildUI];
    
    
}

#pragma mark 搭建UI界面
-(void)buildUI
{
    self.view.backgroundColor = [UIColor whiteColor];
    
    //设置左上角按钮
    UIBarButtonItem *map = [[UIBarButtonItem alloc]initWithTitle:@"北京" style:UIBarButtonItemStylePlain target:self action:@selector(mapAction)];
    self.navigationItem.leftBarButtonItem = map;
    //创建搜索栏
    UISearchBar *searchBar = [[UISearchBar alloc] init];
    searchBar.placeholder = @"输入商户名、地点";
    self.navigationItem.titleView = searchBar;
    
    //增加顶部额外的滚动区域
    self.tableView.contentInset = UIEdgeInsetsMake( kTableBorderWidth, 0, 0, 0);
}

#pragma mark 选择地点
-(void)mapAction{
    DLog(@"定位");
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    // Return the number of sections.
    return 0;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    // Return the number of rows in the section.
    return 0;
}

/*
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:<#@"reuseIdentifier"#> forIndexPath:indexPath];
    
    // Configure the cell...
    
    return cell;
}
*/

/*
// Override to support conditional editing of the table view.
- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
    // Return NO if you do not want the specified item to be editable.
    return YES;
}
*/

/*
// Override to support editing the table view.
- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
    if (editingStyle == UITableViewCellEditingStyleDelete) {
        // Delete the row from the data source
        [tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
    } else if (editingStyle == UITableViewCellEditingStyleInsert) {
        // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
    }   
}
*/

/*
// Override to support rearranging the table view.
- (void)tableView:(UITableView *)tableView moveRowAtIndexPath:(NSIndexPath *)fromIndexPath toIndexPath:(NSIndexPath *)toIndexPath {
}
*/

/*
// Override to support conditional rearranging of the table view.
- (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)indexPath {
    // Return NO if you do not want the item to be re-orderable.
    return YES;
}
*/

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Freedom/大众点评/发现/DZDiscoverController.swift
//
//  DZDiscoverController.swift
//  Freedom
//
//  Created by htf on 2018/5/17.
//  Copyright © 2018年 薛超. All rights reserved.
//

import UIKit

class DZDiscoverController: DianpingBaseViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    func buildUI() {
        let searchBar = UISearchBar()
        searchBar.placeholder = "输入商户名、地点"
        navigationItem?.titleView = searchBar
        view.backgroundColor = UIColor.white
        let map = UIBarButtonItem(title: "北京", style: .plain, actionBlick: {() -> Void in
        })
        navigationItem?.leftBarButtonItem = map
        let titles = ["精选", "嗨周末", "变漂亮", "潮餐厅", "出去浪", "探店报告"]
        let controllers = ["DZDealListViewController", "DZDealListViewController", "DZDealListViewController", "DZDealListViewController", "DZDealListViewController", "DZDealListViewController"]
        contentScrollView = BaseScrollOCView.sharedContentTitleView(withFrame: CGRect(x: 0, y: 0, width: APPW, height: APPH - 55), titles: titles, controllers: controllers, inViewController: self)
        contentScrollView.selectBlock = {(_ index: Int, _ dict: [AnyHashable: Any]?) -> Void in
            DLog("点击了%ld,%@", index, dict)
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
